### DISCUSSION TITLE
Is there a way to generate backend client based on a swagger?

### DISCUSSION THREAD

**Discussion by vinclou** [refinedev/refine#2247] [Q&A] [answered]
What I mean is something similar to this tool [openapi-generator-cli](https://github.com/OpenAPITools/openapi-generator-cli). If not what's the current workflow of defining types and interfaces for the rest client?

Thank you so much, for any insights!

**Comment by omeraplak** [comment, marked as answer]
Hey @vinclou ,
Thank you for contacting us! I haven't tried it but will this project work? 
https://www.npmjs.com/package/swagger-typescript-api

I know that's not what you want but similar solution for GraphQL is here,
https://github.com/pankod/refine/issues/2196
